metasync.js

```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
   if (len > 0) next();
   else done();
 };
+
+
+// Asynchrous filter
+// filter :: (a -> Boolean) -> [a] -> ([a] -> Void)
+metasync.filter = function(predicate, array, done) {
```

Comment:
it's JavaScript way `Array.filter(predicate)`, see http://caolan.github.io/async/docs.html and other asynchronous libs do so